### PR TITLE
15.0 mig base exception fix

### PR DESCRIPTION
**Pull request by damdam-s**
tentative to fix #2279

### CODE REVIEW

**Comment by damdam-s** [comment]
superseeds #2279 

splits base_exception module in 2 modules: 

- base_exception for the behaviour
- test_base_exception for the tests

I did that because model loading "on the fly" does not work

**Review by nicomacr — APPROVED** [review]
LGTM

**Review by ivantodorovich — APPROVED** [review]

**Comment by mylbco** [comment]
When will this be merged ?

**Review by leemannd — APPROVED** [review]

File: test_base_exception/tests/test_base_exception.py [hunk 1]
```diff
@@ -17,29 +14,10 @@
 class TestBaseException(common.TransactionCase):
     @classmethod
     def setUpClass(cls):
-        super(TestBaseException, cls).setUpClass()
-        cls.loader = FakeModelLoader(cls.env, cls.__module__)
-        cls.loader.backup_registry()
-
```
Comment (cesar-tecnativa) [inline test_base_exception/tests/test_base_exception.py]: do not remove tests

File: test_base_exception/tests/test_base_exception.py [hunk 2]
```diff
@@ -17,29 +14,10 @@
...
-            field = fields.Many2many("base.exception.test.purchase")
-            cls.exception_rule._add_field("test_purchase_ids", field)
-            cls.exception_rule._fields["test_purchase_ids"].depends_context = None
-        cls.exception_confirm = cls.env["exception.rule.confirm"]
-        cls.exception_rule._fields["model"].selection.append(
-            ("base.exception.test.purchase", "Purchase Order")
-        )
-
-        cls.exception_rule._fields["model"].selection.append(
-            ("base.exception.test.purchase.line", "Purchase Order Line")
```
Comment (cesar-tecnativa) [inline test_base_exception/tests/test_base_exception.py]: do not remove tests

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

File: test_base_exception/tests/test_base_exception.py (hunk 2 above)
Comment (cesar-tecnativa) [reply test_base_exception/tests/test_base_exception.py]: I did a new PR https://github.com/OCA/server-tools/pull/2366 that solves this problem

**Comment by pedrobaeza** [comment]
Superseded by #2366

**Pull request closed without merging**